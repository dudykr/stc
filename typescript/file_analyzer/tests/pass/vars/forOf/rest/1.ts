// @target: es2015
let array: { x: number, y: string }[];
for (let { x, ...restOf } of array) {
    [x, restOf];
}
let xx: number;
let rrestOff: { y: string };
for ({ x: xx, ...rrestOff } of array) {
    [xx, rrestOff];
}

export { }