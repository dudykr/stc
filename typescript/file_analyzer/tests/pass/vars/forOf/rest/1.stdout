warning: Type
 --> $DIR/tests/pass/vars/forOf/rest/1.ts:3:30
  |
3 | for (let { x, ...restOf } of array) {
  |                              ^^^^^
  |
  = note: {
              x: number;
              y: string;
          }[]

warning: Type
 --> $DIR/tests/pass/vars/forOf/rest/1.ts:4:6
  |
4 |     [x, restOf];
  |      ^
  |
  = note: number

warning: Type
 --> $DIR/tests/pass/vars/forOf/rest/1.ts:4:9
  |
4 |     [x, restOf];
  |         ^^^^^^
  |
  = note: {
              y: string;
          }

warning: Type
 --> $DIR/tests/pass/vars/forOf/rest/1.ts:4:5
  |
4 |     [x, restOf];
  |     ^^^^^^^^^^^
  |
  = note: (number | {
              y: string;
          })[]

warning: Type
 --> $DIR/tests/pass/vars/forOf/rest/1.ts:8:32
  |
8 | for ({ x: xx, ...rrestOff } of array) {
  |                                ^^^^^
  |
  = note: {
              x: number;
              y: string;
          }[]

warning: Type
 --> $DIR/tests/pass/vars/forOf/rest/1.ts:9:6
  |
9 |     [xx, rrestOff];
  |      ^^
  |
  = note: number

warning: Type
 --> $DIR/tests/pass/vars/forOf/rest/1.ts:9:10
  |
9 |     [xx, rrestOff];
  |          ^^^^^^^^
  |
  = note: {
              y: string;
          }

warning: Type
 --> $DIR/tests/pass/vars/forOf/rest/1.ts:9:5
  |
9 |     [xx, rrestOff];
  |     ^^^^^^^^^^^^^^
  |
  = note: (number | {
              y: string;
          })[]

